Automattic/jetpack
# Additional analytics for jetpack recommendations

#### Changes proposed in this Pull Request:

* This PR adds new analytics events for the recommendations flow. See below for event names/ details.

#### Does this pull request change what data or activity we track or use?


Yes, this PR adds 2 new analytics events:
- `jetpack_recommendations_recommendation_viewed` - This event is recorded when a recommended feature or resource step is viewed in the recommendations flow. The name of the step/feature is passed as a prop of the event.
- `recommendations_conditional_recommendation_enabled ` - This event is recorded when the conditions are met to enable a new conditional recommendation. The name of the recommendation is passed as a prop of the event.

#### Testing instructions:






* Check out this branch on your test site, make sure you are connected to Jetpack.
* Navigate to `wp-admin/admin.php?page=jetpack#/recommendations/monitor`
* Filter your network requests by `t.gif`, you should be able to see the `jetpack_recommendations_recommendation_viewed ` sent for the monitor step.
* If you navigate to other steps in the recommendations flow, you should see the same event sent for the other feature steps.

----

* If you have already enabled some conditional recommendations on your test site, it is helpful to reset the `recommendations_conditional` option before completing the next steps. You can do this with the command `wp jetpack options delete recommendations_conditional`. If you are using the docker development environment, you can run this command using `jetpack docker`
* After resetting the conditional recommendations, publish a new post to enable the conditional recommendation for publicize. This should send the `recommendations_conditional_recommendation_enabled ` event. 
* Confirm that this event is received/ recorded.
